PrebidMobile/AdUnits/Native/NativeAd.swift

```diff
@@ -36,7 +36,13 @@ public class NativeAd: NSObject, CacheExpiryDelegate {
     private var viewabilityTimer: Timer?
     private var viewabilityValue = 0
     private var impressionHasBeenTracked = false
-    private weak var viewForTracking: UIView?
+    private weak var viewForTracking: UIView? {
+        didSet {
+            if (viewForTracking == nil) {
+                invalidateTimer(viewabilityTimer)
```

Comment:
In this case, we shouldn't provide an "unregister" functionality. Otherwise, the publisher can just not register the view. And that is it. 

In the case of an unregistered view, the impression event may not be tracked when the ad view appears on the screen. It is the biggest disadvantage.  

Instead, I suggest to review the following aspects:
- invalidation of the timer when the impression is tracked. It is already implemented in the method `trackImpression`. So we just need to make sure that it works and the viewability checker timer is stopped after the impression is tracked. 
- review the timer period. For now, it is really small. And I believe it is because of the bug.

The spec for `scheduledTimer` states 

> /// - parameter:  interval  The number of seconds between firings of the timer. If seconds is less than or equal to 0.0, this method chooses the nonnegative value of 0.1 milliseconds instead.

And we call this method with the following parameter:

```
    private static let nativeAdCheckViewabilityForTrackingFrequency = 0.25
```

It means that the code "wants" to fire the timer four times per second, which doesn't seem bad, but instead, it does it every 0.1 milliseconds, which is bad and unnecessary. 

So I think if we reduce the frequency and make sure that the timer stops after tracking the impression event, the behavior will be correct and will not consume CPU. 

cc @shinwan2

Reply:
Sorry, @jsligh pointed out that the parameter is correct and corresponds to the spec.

I double-checked the behavior in the Internal Test App with an integration case, `Native Add Feed (In-App).` The timer works as expected - checks the viewability 4 times per second. 

It looks like a normal cadence. 

Now, we should be sure that it stops after the impression is tracked.